### DISCUSSION TITLE
API [getListingsByShop] query param [state] has edit?

### DISCUSSION THREAD

**Discussion by IBeanCN** [etsy/open-api#1300] [Q&A] [answered]
When I initiate a query request, if the state is active or inactive, the data in the edit state will be returned.

![image](https://github.com/user-attachments/assets/6fcf502d-63de-4f37-84d4-d2987ba35dcd)

**Comment by franciscodelgadodev** [comment]
We are facing the same problem. Any update from the Etsy team would be helpful.

Thanks

**Comment by will-blanco** [comment]
Same issue here—adding more details below.

When querying GET /v3/application/shops/{shop_id}/listings with state=active or state=draft, the response includes listings whose state is "edit". This also inflates the top-level count field, which we rely on. We expect strict filtering to the requested state only.

**Endpoints affected:**

GET https://openapi.etsy.com/v3/application/shops/{shop_id}/listings?state=active&limit=1

GET https://openapi.etsy.com/v3/application/shops/{shop_id}/listings?state=draft&limit=1
(Also observed with state=sold_out returning edit listings.)

**Reproduction steps:**

Ensure the shop has 0 active listings but ≥1 listings in state="edit" (or any shop with some edit listings).

**Call:**

GET /v3/application/shops/{shop_id}/listings?state=active&limit=1
Headers:
  x-api-key: <REDACTED>
  Authorization: Bearer <REDACTED>


**Actual result:**

Response results[0].state is "edit" (not "active").

Top-level "count" is > 0, reflecting edit items.

Repeat with state=draft (and optionally state=sold_out) and observe the same inclusion of "edit" listings.

**Expected behavior:**

state=active → only listings where state === "active"; "edit" must be excluded.

state=draft → only listings where state === "draft"; "edit" must be excluded.

Top-level "count" should reflect only the requested state.

**Actual behavior:**

"edit" listings are returned in results and included in count when filtering state=active, state=draft.

This produces incorrect counts and misclassification.

**Comment by will-blanco** [comment]
Any update from the Etsy team would be helpful.

**Comment by will-blanco** [comment]
Any update from the Etsy team would be helpful.

[a comment by an automated account is not shown]
